src/main/kotlin/com/bugsnag/android/gradle/internal/BugsnagHttpClientHelper.kt

```diff
@@ -19,33 +36,102 @@ interface BugsnagHttpClientHelper : AutoCloseable {
         okHttpClient.connectionPool.evictAll()
         okHttpClient.cache?.close()
     }
+
+    companion object {
+        fun create(
+            project: Project,
+            bugsnag: BugsnagPluginExtension
+        ): Provider<out BugsnagHttpClientHelper> {
+            val gradleVersion = project.gradle.versionNumber()
+            val providers = project.providers
+            val proxyHost = providers.systemPropertyCompat(KEY_PROXY_HOST, gradleVersion)
+            val proxyPort = providers.systemPropertyCompat(KEY_PROXY_PORT, gradleVersion)
+            val proxyUsername = providers.systemPropertyCompat(KEY_PROXY_USER, gradleVersion)
+            val proxyPassword = providers.systemPropertyCompat(KEY_PROXY_PASSWORD, gradleVersion)
```

Comment:
If you didn’t need system properties, you could have probably just used the JVM’s built-in stuff that uses these properties for you.

```
    OkHttpClient.Builder()
        .proxySelector(ProxySelector.getDefault())
        .proxyAuthenticator(okhttp3.Authenticator.JAVA_NET_AUTHENTICATOR)
        ...
        .build()
```

Reply:
That's a good point! Gradle reaaaaally wants to have all these property lookups run through them for configuration caching and build hermeticity though. Will go with this for now and I'll raise this up with gradle people separately to see how much it matters when they're not being used as task inputs. Easy to change later since this PR makes all these internal APIs now

Reply:
Checked in the gradle slack, looks like it's ok to use this since it doesn't affect configuration of tasks. Will update 👍

Reply:
Updated here 63bb67e. There may still be some potential use if, say, a user wanted to specify these via the plugin extension. I think that seems worth waiting until someone actually asks for it though